uniffi_core/src/ffi_converter_traits.rs

```diff
@@ -245,7 +245,7 @@ pub unsafe trait Lower<UT>: Sized {
 
     /// Convenience method
     fn lower_into_rust_buffer(obj: Self) -> RustBuffer {
-        let mut buf = ::std::vec::Vec::new();
+        let mut buf = ::std::vec::Vec::with_capacity(size_of::<Self>());
```

Comment:
This would be a great optimization, but is `size_of<Self>` the right value?  Our serialization code and Rust's type layout are very different.  Still, maybe `size_of<Self>` is a good estimate -- I'm not really sure.

Can you do some tests?  Like create a set of Records and Enums, some with few fields/variants, some with lots of them, then compare `size_of::<Self>` against `lower_into_rust_buffer(self).len()`.  I'd love to see the results of that.